logstash-core/lib/logstash/bootstrap_check/persisted_queue_config.rb

```diff
@@ -0,0 +1,15 @@
+# encoding: utf-8
+require 'logstash/errors'
+
+module LogStash
+  module BootstrapCheck
+    class PersistedQueueConfig
+      def self.check(settings)
+        return unless settings.get('queue.type') == 'persisted'
+        if settings.get('queue.page_capacity') > settings.get('queue.max_bytes')
+          raise(LogStash::BootstrapCheckError, 'Invalid configuration, `queue.page_capacity` must be smaller or equal to `queue.max_bytes`')
+        end
```

Comment:
should we also check for available disk space in the bootstrap check, instead of waiting until we attempt to create a page?

Reply:
not really because we need to take into account the existing queue data/state and this is computed in queue open. For example if max_bytes is 2GB but there is already 1GB of pages allocation then available disk space must be >= 1GB, not 2.